xarray/backends/zarr.py

```diff
@@ -70,7 +70,14 @@ def __init__(self, variable_name, datastore):
         array = self.get_array()
         self.shape = array.shape
 
-        dtype = array.dtype
+        # preserve vlen string object dtype (GH 7328)
```

Comment:
Same as above, should preserve the metadata in case of vlen - string for zarr. This uses private zarr properties, so an review from an zarr expert would be helpful (ping @jhamman).

Reply:
I am looking... I don't quite follow what's going on here. The empty array has no dtype (defaults to float), so we explicitly make it stringy in the case that the filter was designed for strings? That sounds reasonable to me. Which is the "private property"? Looking at the filter ID is certainly OK.

Reply:
So `array._meta["filters"]` is publilc API?

Reply:
Thank you very much @martindurant. I must have missed `codec_id" while looking for a way to retrieve the wanted information.